lib/include/ert/nexus/nexus_plot.hpp

```diff
@@ -41,17 +42,38 @@ struct unexpected_eof : public std::runtime_error {
 
 class NexusPlot {
 public:
+    struct NexusClassItem {
+        std::string name;
+        int timestep;
+        float time;
+        int num_items;
+        int max_items;
+        int max_perfs;
+        std::map< std::string, float > var;
+
+        float operator[] (std::string s) {
+            return this->var[s];
+        }
+    };
+
+    typedef std::map< std::string, std::vector<NexusClassItem> > timestep_data;
+
     NexusPlot( const std::string& );
     NexusPlot( std::istream& );
 
     int32_t num_classes = 0;
     int32_t day, month, year;
     int32_t nx, ny, nz;
     int32_t ncomp;
-    std::vector<std::string> class_names;
-    std::vector<int32_t> vars_in_class;
-    std::vector< std::vector<std::string> > var_names;
+    std::vector< std::string > class_names;
+    std::map< std::string, int32_t > vars_in_class;
+    std::map< std::string, std::vector<std::string> > var_names;
```

Comment:
unless there is some weirdness hidden somewhere, I'd expect `class_names == va_names.keys()` (no, there is no keys() method, but that's the idea) and also `vars_in_class["class"] = var_names["class"].size()`. So it is sufficient to store the `var_names` map, and you don't have to bother with keeping these 3 variables consistent with each other